sktime/classification/early_classification/tests/_test_teaser.py

```diff
@@ -35,15 +35,22 @@ def test_teaser_on_unit_test_data():
     final_decisions = np.zeros(10)
 
     X_test = from_nested_to_3d_numpy(X_test)
-    states = None
+    X_test = X_test[indices, :, :]
+
+    ec_idx = np.arange(0, 10)
+    decisions = np.zeros(len(X_test), dtype=bool)
     for i in teaser.classification_points:
-        X = X_test[indices, :, :i]
-        probas, decisions, states = teaser.predict_proba(X, state_info=states)
+        inv_dec = np.invert(decisions)
+        X_test = X_test[inv_dec, :, :]
+        ec_idx = ec_idx[inv_dec]
+
+        X = X_test[:, :, :i]
+        probas, decisions, states = teaser.update_predict_proba(X)
 
-        for n in range(10):
-            if decisions[n] and final_decisions[n] == 0:
-                final_probas[n] = probas[n]
-                final_decisions[n] = i
+        for n in range(len(X)):
```

Comment:
isn't this the same as?

`final_decisions = decisions | final_decisions`

Using boolean or between two numpy arrays?

Reply:
and

`final_probas[decisions] = probas`

Reply:
I'm a bit unsure which bit of code you are referring to, sorry.

If its:
```
       for n in range(len(X)):
            if decisions[n] and final_decisions[ec_idx[n]] == 0:
                final_probas[ec_idx[n]] = probas[n]
                final_decisions[ec_idx[n]] = i
```
then the size of decisions is decreasing as cases are successfully classified. The index of decisions will match the input X, but not necessarily the full dataset.

Reply:
Ok, I agree my short version does not help. 

But still: the code is 13 Lines now for the streaming use case. 

```    
    ec_idx = np.arange(0, 10)
    decisions = np.zeros(len(X_test), dtype=bool)
    for i in teaser.classification_points:
        inv_dec = np.invert(decisions)
        X_test = X_test[inv_dec, :, :]
        ec_idx = ec_idx[inv_dec]

        X = X_test[:, :, :i]
        probas, decisions, states = teaser.update_predict_proba(X)

        for n in range(len(X)):
            if decisions[n] and final_decisions[ec_idx[n]] == 0:
                final_probas[ec_idx[n]] = probas[n]
                final_decisions[ec_idx[n]] = i
```

And you have to repeat it every time, you do this. I think, we should think about something much more compact.

Reply:
It is sort of a consequence of us doing it backwards, we already have the full series and need to shorten it instead of the array growing over time so we can't just overwrite X_test. A lot of it is required to collate the results in the correct order, so we can compare later also.

I agree it is not great, if anyone has a suggestion on how to do it better that would be good. I will try to think of a better way. Probably fine for testing, could be better for the example notebook.

Reply:
as a start, what about something like this:


 ```
   for i in teaser.classification_points:
        X = X_test[:, :, :i]
        probas, decisions, states, open_idx, final_idx = teaser.update_predict_proba(X, X_idx)

        final_probas[final_idx] = probas
        final_decisions[final_idx] = decisions

        X_test = X_test[open_idx]
        X_idx = open_idx
```

I.e. pass the indices of the current TS to proba as an optional parameter and in this case return the `open_idx` for ts where no decision was taken and  `final_idx` for ts where decision was taken?